frontends/concrete-python/concrete/fhe/mlir/converter.py

```diff
@@ -618,6 +618,20 @@ def tlu(self, ctx: Context, node: Node, preds: List[Conversion]) -> Conversion:
         variable_input_index = variable_input_indices[0]
         variable_input = preds[variable_input_index]
 
+        if variable_input.origin.properties.get("name") == "truncate_bit_pattern":
+            original_bit_width = variable_input.origin.properties["original_bit_width"]
+            lsbs_to_remove = variable_input.origin.properties["kwargs"]["lsbs_to_remove"]
+            truncated_bit_width = original_bit_width - lsbs_to_remove
+
+            if variable_input.bit_width > original_bit_width:
+                bit_width_difference = variable_input.bit_width - original_bit_width
+                shifter = ctx.constant(
+                    ctx.i(variable_input.bit_width + 1), 2**bit_width_difference
+                )
+                variable_input = ctx.mul(variable_input.type, variable_input, shifter)
+
+            variable_input = ctx.reinterpret(variable_input, bit_width=truncated_bit_width)
```

Comment:
ok so the actual bitwidth reduction is only possible here with the TLU after ?

Reply:
> actual bitwidth reduction
Is an optimization for TLU, it's not the execution semantics so it's not for leveled ops.